instrumentation/reactor-core-3.1/src/main/java8/io/opentelemetry/auto/instrumentation/reactor/core/FluxAndMonoSubscribeAdvice.java

```diff
@@ -33,11 +33,12 @@ public static SpanWithScope methodEnter(
   @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
   public static void methodExit(
       @Advice.Enter final SpanWithScope spanWithScope, @Advice.Thrown final Throwable throwable) {
+    if (spanWithScope == null) {
```

Comment:
when do you end up with null here? Should we have a no-op version of the class for those cases?

Reply:
when the `Advice.OnMethodEnter` method above returns `null`

Reply:
maybe that should return a no-op version, rather than null?

Reply:
this is the pattern used throughout the code, so that's a bigger discussion, also less efficient because then we end up calling setAttribute a bunch of times on no-op Spans unnecessarily

Reply:
maybe this discussion can be part of auto/lib instrumentation convergence

Reply:
oh, i forgot, even in auto/lib convergence, this pattern of passing across SpanWithScope will only exist in auto instrumentation, so this discussion doesn't apply to auto/lib instrumentation convergence, and this will stay internal API only (so i think ok to continue using `null`?)

Reply:
Just my opinion, which you are free to ignore, but using `null` as the contract between these methods seems like a dangerous anti-pattern, because (as you see in this very PR), it's easy to forget to check for nulls. I'm not often a fan of the null object pattern, but this sure seems like a good case for it. no-op methods are very fast and will get inlined quickly by the JIT if they get called much.

Reply:
the calling of no-op methods will get inlined, but not the calculations of the attribute values, which can be expensive (e.g. calling Throwable.printStackTrace())

Reply:
If `finishSpanIfPresent` checked for `span.isInvalid()` then that call could also be avoided.

Reply:
this pattern (passing null from `OnMethodEnter` to `OnMethodExit`) is used throughout the existing codebase, so i think better for this PR to stay consistent, and if we want to change the pattern then that belongs to a different PR where we can look at the change in the context of the full codebase